contents/tutorials/flutter-ab-tests.md

```diff
@@ -0,0 +1,356 @@
+
+  const FeatureScreenView({Key? key, required this.isTestVariant}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: isTestVariant ? Colors.green : Colors.red,
+      body: Center(
+        child: ElevatedButton(
+          child: const Text('Click Me!'),
+          onPressed: () {
+            // Event capturing will go here
+          },
+        ),
+      ),
+    );
+  }
+}
+```
+
+Press **F5** and run the app in any emulator (we chose Android) to see your app in action.
+
+![Basic setup of the Flutter app](../images/tutorials/flutter-ab-tests/basic-app.mp4)
+
+## 2. Add PostHog to your app
+
+With our app set up, it’s time to install and set up PostHog. If you don't have a PostHog instance, you can [sign up for free](https://us.posthog.com/signup).
+
+To start, install [PostHog’s Flutter SDK](/libraries/flutter) by adding `posthog_flutter` to your `pubspec.yaml`:
+
+```yaml file=pubspec.yaml
+# rest of your code
+
+dependencies:
+  flutter:
+    sdk: flutter
+  posthog_flutter: ^4.0.1
+
+# rest of your code
+```
+
+Next, we configure PostHog using our project API key and instance address. You can find these in [your project settings](https://us.posthog.com/settings/project).
+
+### Android setup
+
+For Android, add your PostHog configuration to your `AndroidManifest.xml` file located in the `android/app/src/main`:
+
+```xml file=android/app/src/main/AndroidManifest.xml
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="your.package.name">
+    <application>
+        <!-- ... other configuration ... -->
+        <meta-data android:name="com.posthog.posthog.API_KEY" android:value="<ph_project_api_key>" />
+        <meta-data android:name="com.posthog.posthog.POSTHOG_HOST" android:value="<ph_instance_address>" /> <!-- usually 'https://app.posthog.com' or 'https://eu.posthog.com' -->
+        <meta-data android:name="com.posthog.posthog.TRACK_APPLICATION_LIFECYCLE_EVENTS" android:value="true" />
+        <meta-data android:name="com.posthog.posthog.DEBUG" android:value="true" />
+    </application>
+</manifest>
+```
+
+You'll also need to update the minimum Android SDK version to `21` in `android/app/build.gradle`:
+
+```gradle_kotlin file=android/app/build.gradle
+// rest of your config
+
+    defaultConfig {
+      minSdkVersion 21
+      // rest of your config
+    }
+
+// rest of your config
+```
+
+You may also need to ensure you're running on the latest Kotlin version. To do this, go to the [Kotlin website](https://kotlinlang.org/docs/releases.html#release-details) to find the latest version number. Then set this version number in the `build.gradle` in the root of your `android` directory (and not the `build.gradle` located in `android/app`):
+
+```js file=android/build.gradle
+buildscript {
+    ext.kotlin_version = '1.9.22' // use the latest Kotlin version number here
+    // rest of config
+```
+
+### iOS setup
```

Comment:
Maybe for future ones we just set up for Android and say if you want the others you can see the docs?

Reply:
hmm will give it some thought